motion/motion.go

```diff
@@ -194,6 +222,25 @@ func (d *motionDetector) warmerDiffFrames(a, b, out *lepton3.Frame) *lepton3.Fra
 	return out
 }
 
+func (d *motionDetector) updateBackground(new_frame *lepton3.Frame) (float64, bool) {
+	if d.count == 0 {
+		d.background = new_frame.Pix
+		return 0, true
+	}
+	var changed bool = false
+	var sum float64 = 0
+	for y := d.start; y < d.rowStop; y++ {
+		for x := d.start; x < d.columnStop; x++ {
+			if uint16(float32(new_frame.Pix[y][x])*d.backgroundWeight) < d.background[y][x] {
+				d.background[y][x] = new_frame.Pix[y][x]
+				changed = true
+			}
+			sum = sum + float64(d.background[y][x])
```

Comment:
This is unsafe as the sum can easily be larger than the largest float64 value.

You'll need to use a cumulative average.  You need a float64 for the current average and an integer counter which is incremenet for each value (pixel in this case).

Pseudo-code:
```
average = average + ((value - average) / count)
```